driver/remote/factory.go

```diff
@@ -0,0 +1,109 @@
+package remote
+
+import (
+	"context"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/buildx/driver"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+const priority = 90
+
+func init() {
+	driver.Register(&factory{})
+}
+
+type factory struct {
+}
+
+func (*factory) Name() string {
+	return "remote"
+}
+
+func (*factory) Usage() string {
+	return "remote"
+}
+
+func (*factory) Priority(ctx context.Context, api dockerclient.APIClient) int {
+	return priority
```

Comment:
I think this could use a bit of logic change. You could update the signature of the `Priority()` method to take in the endpoint/args value that you can pass to the `GetDefaultFactory()` . If it is set to a tcp/tls address (or `BUILDKIT_HOST` set) then this priority in here would be higher(below 30).

Ideally, we could also detect that `docker-container://` and `kube-pod://` addresses go to the correct driver but that can be a follow-up.